code/script.js

```diff
@@ -225,101 +173,213 @@ const generateBoard = () => {
-  } else {
+
+    // for accessories and other
+    if (category === "accessories"|| category === "other")
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
     }
+
+
+    // Invoke a function to redraw the board with the remaining people.
+    generateBoard();
+   
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+ 
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+
+
+  if (confirm (`Are you sure you want to make a guess on ${personToConfirm}?`)){
+    const playerGuess = personToConfirm
+    checkMyGuess(playerGuess)
+  }
+
 }
 
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
+const checkMyGuess = (personToConfirm) => {
   // 1. Check if the personToCheck is the same as the secret person's name
+   if ( secret.name === personToConfirm) {
+     winOrLoseText.innerHTML = `YOU WON is ${secret.name}!`
+    }
+
   // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+  else {
+    winOrLoseText.innerHTML = `GAME OVER!` 
+   }
+// 3. Show the win or lose section
+   winOrLose.style.display = "flex"
+// 4. Hide the game board
+   board.style.display = "none"
+   aside.style.display = "none"
+
+  }
+
+  
+
+//timer
+
+const startingMinutes = 5;
+let time = startingMinutes * 60;
+
+setInterval(function () {
+const minutes = Math.floor(time / 60);
+let seconds = time % 60;
+
+seconds = seconds < 5 ? '0' + seconds : seconds;
+
+countdownEl.innerHTML = `${minutes}:${seconds}`;
+time--;
+}, 1000);
+
+
+  // background song option
+
+ const muteSound = () => {
+   if (backGroundSound.muted === true) {
+     backGroundSound.muted = false
+     backGroundSound.play()
```

Comment:
Perhaps a backGroundSound.loop = true to make the music keep going unless the player mutes it again?